static/js/containers/pages/login/LoginEmailPage.js

```diff
@@ -21,27 +21,25 @@ type Props = {
   loginEmail: (email: string, next: ?string) => Promise<Response<AuthResponse>>
 }
 
-class LoginEmailPage extends React.Component<Props> {
-  async onSubmit({ email }, { setSubmitting, setErrors }) {
+export class LoginEmailPage extends React.Component<Props> {
+  async onSubmit(
+    { email }: EmailFormValues,
+    { setSubmitting, setErrors }: any
+  ) {
     const { loginEmail, location, history } = this.props
     const nextUrl = getNextParam(location.search)
 
-    /* eslint-disable camelcase */
     try {
-      const result = await loginEmail(email, nextUrl)
-      const { state, errors } = result.transformed.auth
+      const { body } = await loginEmail(email, nextUrl)
 
-      if (state === STATE_LOGIN_PASSWORD) {
-        history.push(routes.login.password)
-      } else if (errors.length > 0) {
-        setErrors({
-          email: errors[0]
-        })
-      }
+      handleAuthResponse(history, body, {
+        // eslint-disable-next-line camelcase
+        [STATE_ERROR]: ({ field_errors }: AuthResponse) =>
+          setErrors(field_errors)
```

Comment:
Take or leave: it would feel a bit arbitrary/hacky to just pass `setErrors` into `handleAuthResponse`, but this same handler is being defined and passed in 5 times in this PR, and I think it's reasonable to assume that any invocation of `handleAuthResponse` that (a) has a `STATE_ERROR` response state, and (b) is being invoked as part of a form submission is going to want to do this. Something feels off about the fact that you need to redefine if you want field errors from the auth response to be set properly in the state. Maybe we could define a thin wrapper around `handleAuthResponse` called `handleFormSubmitAuthResponse` or something? Again, I could go either way on this

Reply:
I had thought about this myself, but I think I landed on not wanting to leak component implementation details into this method, since `setErrors` eventually calls `setState` on the inner `formik` form. For similar reasons I also kept `addUserNotification` calls out of there.

I also just added another case [here](https://github.com/mitodl/mitxpro/pull/853/files#diff-7346aaa1034d79092164e260485b05e9R44-R45) where I'm also calling `setErrors` but for a non-`STATE_ERROR` case (the regression that @mbertrand caught), so maybe that makes this abstraction feel a bit better?